ci: skip tests involving L7 proxy if race detector is enabled

tklauser opened the pull request:
Running tests involving the L7 proxy fail when using the race detector, so skip them in that case.

See individual commit for details.

Fixes #13753 
Fixes #13757

aanm reviewed (approved):
Nice, really simple :D

tklauser committed:
test: detect RACE env var at runtime

This allows to determine whether the Cilium binaries under test were
built using the race detector enabled and the tests should be run
skipping any tests that fail with the race detector enabled.

Signed-off-by: Tobias Klauser <tklauser@distanz.ch>

tklauser committed:
ci: skip kafka and cassandra runtime tests if race detector enabled

Running tests involving proxylib fail when using the race detector, so
skip them in that case.

Fixes #13753

Signed-off-by: Tobias Klauser <tklauser@distanz.ch>

tklauser committed:
ci: skip k8s Kafka/L7 tests if race detector is enabled

K8s tests involving the L7 proxy fail when using the race detector, so
skip them in that case.

Fixes #13757

Signed-off-by: Tobias Klauser <tklauser@distanz.ch>

christarazi commented on Documentation/contributing/testing/e2e.rst:
```diff
@@ -227,7 +229,7 @@ framework in the ``test/`` directory and interact with ginkgo directly:
             Specifies timeout for test run (default 24h0m0s)
 
-    Ginkgo ran 1 suite in 4.312100241s
+    Ginkgo ran 1 suite in 3.086476005s
```
Unnecessary change?

tklauser replied:
Side effect of copy&pasting the output :smile: I'll drop it if another revision is needed.

tklauser replied:
Dropped the flag again as it's not necessary, we can just check the `RACE` env var at runtime.

christarazi reviewed (approved):
Nice. Just one non-blocking comment.
